llrpConnector/src/main/java/io/vantiq/extsrc/llrpConnector/LLRPConnector.java

```diff
@@ -0,0 +1,1546 @@
+
+    private Socket connection;				// Socket connection to reader
+    private DataOutputStream out;			// Stream when writing to reader
+    private ReadThread rt = null;			// Reads LLRP msg and add to queue
+    private Thread msgThread = null;		// Process regular msgs from reader
+    private Thread tagThread = null;		// Process tag data msgs from reader
+    private Thread readerAliveThread = null;	// Identify when reader is offline
+    private LLRPConnectorCore llrpCore;
+    private Boolean readerOffline = true;	// Indicates the Reader is Offline
+    private Boolean vantiqSourceOffline = true;	// Indicates the connection to the Vantiq Source is Offline
+    Instant lastKeepAlive = Instant.now();	// Last point we rcvd from reader
+
+    /*
+     * RECONNECT_INTERVAL is the duration to wait to check if Reader is online.
+     * It may take 15-30 seconds to flush through stored messages on the reader
+     * since we do not turn off the reading of tags (stored on reader and
+     * processed on reconnect).
+     * The KEEPALIVE_INTERVAL is the periodic interval the reader will use
+     * to send a KEEPALIVE message.  If we don't receive a KEEPALIVE message
+     * within the RECONNECT_INTERVAL, we drop the connection and try to
+     * reconnect indefinitely.
+     */
+    final static int KEEPALIVE_INTERVAL = 5000;
+    final static int RECONNECT_INTERVAL = 10000;
+
+    /**
+     * Message queues for the tags and other messages
+     */
+    private LinkedBlockingQueue<LLRPMessage> queue
+            = new LinkedBlockingQueue<LLRPMessage>();
+    private LinkedBlockingQueue<LLRPMessage> tagQueue
+            = new LinkedBlockingQueue<LLRPMessage>();
+
+    /**
+     * Current reader Capabilities and Configuration information.
+     * Captured but not used directly for now.
+     */
+    private JSONObject readerCapabilities = null;
+    private JSONObject readerConfiguration = null;
+    private String readerId = "reader1";
+
+
+
+    /**
+     * The constructor creates a connection to the reader and sends LLRP
+     * messages. Once connected, ideal communication is as follows:
+     *     From Enterprise Connector	From the Reader
+     *  1) GET_READER_CAPABILIITIES 	GET_READER_CAPABILITIES_RESPONSE
+     *  2) SET_READER_CONFIG 			SET_READER_CONFIG_RESPONSE
+     *  3) GET_READER_CONFIG 			GET_READER_CONFIG_RESPONSE
+     *  4) ADD_ROSPEC				 	ADD_ROSPEC_RESPONSE
+     *  5) ENABLE_ROSPEC				ENABLE_ROSPEC_RESPONSE
+     *  6) ENABLE_EVENTS_AND_REPORTS	(NONE)
+     *
+     * During the above cycle, all other READER_EVENT_NOTIFICATION messages
+     * are ignored.  After the reader starts up, it queues all messages
+     * and processes them asynchronously.
+     *
+     * A KEEPALIVE message is configured to be sent from the reader
+     * periodically and the LLRP Enterprise Connector sends back a
+     * KEEPALIVE_ACK message to the reader.  If this ACK is not sent, the
+     * reader will shutdown.
```

Comment:
_shutdown_ means _stop sending_?  Or power off?

Reply:
changed to 'disconnect'